src/wrapper.js

```diff
@@ -0,0 +1,162 @@
+/* globals WalkMeAPI, _walkMe */
+'use strict';
+
+const Promise = require('es6-promise').Promise;
+const assign = require('lodash/assign');
+const find = require('lodash/find');
+const findLast = require('lodash/findLast');
+const sortBy = require('lodash/sortBy');
+
+const wrapper = {};
+const API_LOAD_INTERVAL = 500;
+const API_LOAD_ATTEMPTS = 120;
+let isLoadedPromise;
+
+const StepType = {
+  Step: 0,
+  PopupStep: 1,
+  SuperStep: 2
+};
+
+wrapper.Event = {
+  AfterMenuClose: 'AfterMenuClose',
+  AfterMenuOpen: 'AfterMenuOpen',
+  BBCodeLinkClicked: 'BBCodeLinkClicked',
+  BackButtonClicked: 'BackButtonClicked',
+  BalloonCreated: 'BalloonCreated',
+  BeforeMenuClose: 'BeforeMenuClose',
+  BeforeMenuOpen: 'BeforeMenuOpen',
+  DoneButtonClicked: 'DoneButtonClicked',
+  LauncherClicked: 'LauncherClicked',
+  LauncherIsShown: 'LauncherIsShown',
+  NextButtonClicked: 'NextButtonClicked',
+  NextStepShown: 'NextStepShown',
+  PlayerInitComplete: 'PlayerInitComplete',
+  SurveyPlayed: 'SurveyPlayed',
+  SurveyCompleted: 'SurveyCompleted',
+  TaskCompleted: 'TaskCompleted',
+  TasksLoaded: 'TasksLoaded',
+  UserStoppedWalkthru: 'UserStoppedWalkthru',
+  UserStoppedWalkthruAfterStop: 'UserStoppedWalkthruAfterStop',
+  WalkthruAborted: 'WalkthruAborted',
+  WalkthruFailedToStart: 'WalkthruFailedToStart',
+  WalkthruSelected: 'WalkthruSelected'
+};
+
+let afterApiLoad = function afterApiLoad (fn) {
+  if (!isLoadedPromise) {
+    isLoadedPromise = new Promise(function isLoaded (resolve, reject) {
```

Comment:
There is no failure handler here or on line 63 so we're implicitly delegating to the browser to throw, which i think is why we are seeing an Uncaught promise error.

Reply:
this is... yes, good catch!